optuna/importance/_fanova.py

```diff
@@ -0,0 +1,128 @@
+from collections import OrderedDict
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from optuna._experimental import experimental
+from optuna.distributions import BaseDistribution
+from optuna.distributions import CategoricalDistribution
+from optuna.distributions import DiscreteUniformDistribution
+from optuna.distributions import IntUniformDistribution
+from optuna.distributions import LogUniformDistribution
+from optuna.distributions import UniformDistribution
+from optuna.importance._base import BaseImportanceEvaluator
+from optuna.importance._base import get_distributions
+from optuna.importance._base import get_study_data
+from optuna.study import Study
+
+try:
+    from ConfigSpace import ConfigurationSpace
+    from ConfigSpace.hyperparameters import CategoricalHyperparameter
+    from ConfigSpace.hyperparameters import Hyperparameter
+    from ConfigSpace.hyperparameters import UniformFloatHyperparameter
+    from ConfigSpace.hyperparameters import UniformIntegerHyperparameter
+    from fanova import fANOVA
+
+    _available = True
+except ImportError as e:
+    ConfigurationSpace = None
+    Hyperparameter = None
+    fANOVA = None
+
+    _import_error = e
+    _available = False
+
+
+@experimental("1.3.0")
+class FanovaImportanceEvaluator(BaseImportanceEvaluator):
+    """fANOVA parameter importance evaluator.
+
+    .. note::
+
+        Requires the `fanova <https://github.com/automl/fanova>`_ Python package.
+
+    .. seealso::
+
+        `An Efficient Approach for Assessing Hyperparameter Importance
+        <http://proceedings.mlr.press/v32/hutter14.html>`_.
+    """
+
+    def __init__(self) -> None:
+        _check_fanova_availability()
+
+    def evaluate(self, study: Study, params: Optional[List[str]]) -> Dict[str, float]:
+        distributions = get_distributions(study, params)
+        params_data, values_data = get_study_data(study, distributions)
+
+        evaluator = fANOVA(
+            X=params_data,
+            Y=values_data,
+            config_space=_get_configuration_space(distributions),
+            max_features=max(1, int(params_data.shape[1] * 0.7)),
```

Comment:
I'm just curious why you adopted this expression instead of using the default value of `max_features`.

Reply:
It's to handle cases where `X` only has a single or no column (either or both, sorry I can't recall but could confirm). An alternative would be to check the shape of `X` and do an early return or similar.

Reply:
I see. Thanks!